hack/ci-e2e-kind-upgrade.sh

```diff
@@ -125,7 +125,16 @@ function upgrade_to_next_release() {
 
 function set_gardener_upgrade_version_env_variables() {
   if [[ -z "$GARDENER_PREVIOUS_RELEASE" ]]; then
-    export GARDENER_PREVIOUS_RELEASE="$(curl -s https://api.github.com/repos/gardener/gardener/releases/latest | grep tag_name | cut -d '"' -f 4)"
+    previous_minor_version=$(echo "$VERSION" | awk -F. '{printf("%s.%d.*\n", $1, $2-1)}')
+    # List all the tags that match the previous minor version pattern
+    tag_list=$(git tag -l "$previous_minor_version")
+
+    if [ -z "$tag_list" ]; then
+      echo "No tags found for the previous minor version ($VERSION) to upgrade Gardener." >&2
+      exit 1
+    fi
+      # Find the most recent tag for the previous minor version
+      export GARDENER_PREVIOUS_RELEASE=$(echo "$tag_list" | tail -n 1)
```

Comment:
You could improve the indentation:
```suggestion
    # Find the most recent tag for the previous minor version
    export GARDENER_PREVIOUS_RELEASE=$(echo "$tag_list" | tail -n 1)
```